src/ai-bundle/src/Command/ChatCommand.php

```diff
@@ -0,0 +1,213 @@
+    public function __construct(
+        private readonly ServiceLocator $agents,
+    ) {
+        parent::__construct();
+    }
+
+    public function complete(CompletionInput $input, CompletionSuggestions $suggestions): void
+    {
+        if ($input->mustSuggestArgumentValuesFor('agent')) {
+            $suggestions->suggestValues($this->getAvailableAgentNames());
+        }
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->addArgument('agent', InputArgument::OPTIONAL, 'The name of the agent to chat with')
+            ->setHelp(
+                <<<'HELP'
+                The <info>%command.name%</info> command allows you to chat with different agents.
+
+                Usage:
+                  <info>%command.full_name% [<agent_name>]</info>
+
+                Examples:
+                  <info>%command.full_name% wikipedia</info>
+
+                If no agent is specified, you'll be prompted to select one interactively.
+                
+                The chat session is interactive. Type your messages and press Enter to send.
+                Type 'exit' or 'quit' to end the conversation.
+                HELP
+            );
+    }
+
+    protected function interact(InputInterface $input, OutputInterface $output): void
+    {
+        // Skip interaction in non-interactive mode
+        if (!$input->isInteractive()) {
+            return;
+        }
+
+        $agentArg = $input->getArgument('agent');
+
+        // If agent is already provided and valid, nothing to do
+        if ($agentArg) {
+            return;
+        }
+
+        $availableAgents = $this->getAvailableAgentNames();
+
+        if (0 === \count($availableAgents)) {
+            throw new InvalidArgumentException('No agents are configured.');
+        }
+
+        $question = new ChoiceQuestion(
+            'Please select an agent to chat with:',
+            $availableAgents,
+            0
+        );
+        $question->setErrorMessage('Agent %s is invalid.');
+
+        /** @var \Symfony\Component\Console\Helper\QuestionHelper $helper */
+        $helper = $this->getHelper('question');
+        $selectedAgent = $helper->ask($input, $output, $question);
+
+        $input->setArgument('agent', $selectedAgent);
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        // Initialization will be done in execute() after interact() has run
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        // Initialize agent (moved from initialize() to execute() so it runs after interact())
+        $availableAgents = array_keys($this->agents->getProvidedServices());
+
+        if (0 === \count($availableAgents)) {
+            throw new InvalidArgumentException('No agents are configured.');
+        }
+
+        $agentArg = $input->getArgument('agent');
+        $this->agentName = \is_string($agentArg) ? $agentArg : '';
```

Comment:
Why using instance properties (making the service stateful) when they are only used locally ? You should use local variables instead.